MapboxNavigation/RouteController.swift

```diff
@@ -129,7 +129,7 @@ extension RouteController: CLLocationManagerDelegate {
         if let finalHeading = routeProgress.currentLegProgress.upComingStep?.finalHeading {
             let finalHeadingNormalized = wrap(finalHeading, min: 0, max: 360)
             let userHeadingNormalized = wrap(location.course, min: 0, max: 360)
-            courseMatchesManeuverFinalHeading = abs(finalHeadingNormalized - userHeadingNormalized) <= RouteControllerMaximumAllowedDegreeOffsetForTurnCompletion
+            courseMatchesManeuverFinalHeading = abs((finalHeadingNormalized - userHeadingNormalized)) <= RouteControllerMaximumAllowedDegreeOffsetForTurnCompletion || abs((finalHeadingNormalized - userHeadingNormalized) - 360) <= RouteControllerMaximumAllowedDegreeOffsetForTurnCompletion
```

Comment:
Conceptually, you want to determine the difference between two angles, then check that difference against a threshold. There are some great solutions in http://stackoverflow.com/questions/1878907/the-smallest-difference-between-2-angles and http://stackoverflow.com/questions/12234574/calculating-if-an-angle-is-between-two-angles that are provably correct. Some use the `%` operator, while others use trigonometric functions.